src/goto-instrument/goto_instrument_parse_options.cpp

```diff
@@ -1457,6 +1458,46 @@ void goto_instrument_parse_optionst::instrument_goto_program()
       *message_handler);
   }
 
+  // aggressive slicer
+  if(cmdline.isset("aggressive-slice"))
+  {
+    do_indirect_call_and_rtti_removal();
+
+    status() << "Slicing away initializations of unused global variables"
+             << eom;
+    slice_global_inits(goto_model);
+
+    status() << "Performing an aggressive slice" << eom;
+    aggressive_slicert aggressive_slicer(goto_model, get_message_handler());
+
+    if(cmdline.isset("aggressive-slice-call-depth"))
+      aggressive_slicer.call_depth =
+        safe_string2unsigned(cmdline.get_value("aggressive-slice-call-depth"));
+
+    if(cmdline.isset("aggressive-slice-preserve-function"))
```

Comment:
I don't quite understand the use of having this option plus `aggressive-slice-preserve-functions-containing`. Why not allow a regex to specify what should be preserved?

Reply:
I'm saying that there are standard ways of specifying name filtering (wild cards, Regex). 'contains name snippet' is not among these and may confuse users. Using regexes would even save you a couple of lines in the implementation.